test/new-e2e/tests/agent-metric-logs/linux-log-agent/01_file_tailing_test.go

```diff
@@ -0,0 +1,154 @@
+
+	e2e.Run(t, s, logsExampleStackDef())
+}
+
+func (s *LinuxVMFakeintakeSuite) BeforeTest(_, _ string) {
+	// Flush server and reset aggregators before the test is ran
+	s.cleanUp()
+}
+
+func (s *LinuxVMFakeintakeSuite) TearDownSuite() {
+	// Flush server and reset aggregators after the test is ran
+	s.cleanUp()
+}
+
+func (s *LinuxVMFakeintakeSuite) TestLinuxLogTailing() {
+	// Run test cases
+	s.Run("LogCollection", s.LogCollection)
+
+	s.Run("LogPermission", s.LogPermission)
+
+	s.Run("LogRotation", s.LogRotation)
+}
+
+func (s *LinuxVMFakeintakeSuite) LogCollection() {
+	t := s.T()
+	fakeintake := s.Env().Fakeintake
+
+	// Create a new log file
+	s.Env().VM.Execute("sudo touch /var/log/hello-world.log")
+
+	// Part 1: Ensure no logs are present in fakeintake
+	s.EventuallyWithT(func(c *assert.CollectT) {
+		logs, err := fakeintake.FilterLogs("hello")
+		if !assert.NoError(c, err, "Unable to filter logs by the service 'hello'.") {
+			return
+		}
+		// If logs are found, print their content for debugging
+		if !assert.Empty(c, logs, "Logs were found when none were expected.") {
+			cat, _ := s.Env().VM.ExecuteWithError("cat /var/log/hello-world.log")
+			t.Logf("Logs detected when none were expected: %v", cat)
+		}
+	}, 5*time.Minute, 10*time.Second)
+
+	// Part 2: Adjust permissions of new log file
+	_, err := s.Env().VM.ExecuteWithError("sudo chmod 777 /var/log/hello-world.log")
+	assert.NoError(t, err, "Unable to adjust permissions for the log file '/var/log/hello-world.log'.")
+
+	// Generate log
+	generateLog(s, "hello-world")
+
+	// Part 3: Assert that logs are found in intake after generation
+	checkLogs(s, "hello", "hello-world")
+}
+
+func (s *LinuxVMFakeintakeSuite) LogPermission() {
+
+	// Part 4: Block permission and check the Agent status
+	s.Env().VM.Execute("sudo chmod 000 /var/log/hello-world.log")
+	s.EventuallyWithT(func(c *assert.CollectT) {
+		// Check the Agent status
+		statusOutput, err := s.Env().VM.ExecuteWithError("sudo datadog-agent status | grep -A 10 'custom_logs'")
+		if !assert.NoError(c, err, "Issue running agent status: %s", err) {
+			return
+		}
+
+		if !assert.NotContainsf(c, statusOutput, "Status: OK", "Expecting log file to be inaccessible but it is accessible instead: \n %s", statusOutput) {
+			return
+		}
+
+		assert.Contains(c, statusOutput, "denied", "Log file is correctly inaccessible")
+	}, 3*time.Minute, 10*time.Second)
+
+	// Part 5: Restore permissions
+	s.Env().VM.Execute("sudo chmod 777 /var/log/hello-world.log")
+
+	// Part 6: Generate new logs and ceck the Agent status
+	generateLog(s, "hello-world")
+
+	s.EventuallyWithT(func(c *assert.CollectT) {
+		statusOutput, err := s.Env().VM.ExecuteWithError("sudo datadog-agent status | grep -A 10 'custom_logs'")
+		assert.NoError(c, err, "Issue running agent status: %s", err)
+		assert.Contains(c, statusOutput, "Status: OK", "Expecting log file to be accessible but it is inaccessible instead")
+	}, 5*time.Minute, 2*time.Second)
```

Comment:
we should have a Part 7 that checks if the log actually made it to fake-intake, shouldn't we?